python/lsst/ts/MTAlignment/alignment_csc.py

```diff
@@ -144,191 +188,359 @@ async def configure(self, config):
-            data.target_A,
-            data.collection_B,
-            data.pointgroup_B,
-            data.target_B,
+        assert self.model is not None
+        point_delta = await self.model.get_point_delta(
+            p1collection=data.collection_A,
+            p1group=data.pointgroup_A,
+            p1=data.target_A,
+            p2group=data.pointgroup_B,
+            p2=data.target_B,
+            p2collection=data.collection_B,
         )
 
-    async def do_setReferenceGroup(self, data):
-        """Nominal point group to locate tracker station to and provide data
-        relative to"""
+        self.log.debug(f"Point delta: {point_delta}.")
+        # TODO: Publish event
+
+    async def do_setReferenceGroup(self, data: salobj.BaseDdsDataType) -> None:
+        """Set the reference group with respect to which all measumentes will
+        be made against.
+
+        Parameters
+        ----------
+        data : ``cmd_setReferenceGroup.DataType``
+            Command data.
+        """
         self.assert_enabled()
-        await self.model.set_reference_group(data.pointgroup)
+        assert self.model is not None
+        await self.model.set_reference_group(data.referenceGroup)
+
+        # TODO: Publish reference group
 
-    async def do_setWorkingFrame(self, data):
-        """attempt to set the passed string as the SpatialAnalyzer working
-        frame"""
+    async def do_setWorkingFrame(self, data: salobj.BaseDdsDataType) -> None:
+        """Set the SpatialAnalyzer working frame.
+
+        Parameters
+        ----------
+        data : ``cmd_setWorkingFrame.DataType``
+            Command data.
+        """
         self.assert_enabled()
+        assert self.model is not None
         await self.model.set_working_frame(data.workingFrame)
 
-    async def do_halt(self, data):
-        """halts any executing measurement plan and returns to ready state"""
+        # TODO: Publish an event with the working frame.
+
+    async def do_halt(self, data: salobj.BaseDdsDataType) -> None:
+        """Halts any executing measurement plan and returns to ready state.
+
+        Parameters
+        ----------
+        data : ``cmd_halt.DataType``
+            Command data.
+        """
         self.assert_enabled()
+        assert self.model is not None
         await self.model.halt()
 
-    async def do_loadSATemplateFile(self, data):
-        """SA Template file path and name. This is in the filesystem on the
-        T2SA host."""
+    async def do_loadSATemplateFile(self, data: salobj.BaseDdsDataType) -> None:
+        """Load SA Template file.
+
+        Parameters
+        ----------
+        data : ``cmd_loadSATemplateFile.DataType``
+            Command data.
+        """
         self.assert_enabled()
+        assert self.model is not None
         await self.model.load_template_file(data.file)
 
-    async def do_measureDrift(self, data):
-        """measure tracker drift"""
+        # TODO: Publish something?
```

Comment:
I am not really sure what can be published after measuring drift. It is not clear from the API there is something that is actually publishable... I asked Dave, and I'll add if there's something, otherwise we should make sure we add it to the API.